### PR TITLE
Feature/rebase with develop

### PR DESCRIPTION
**Pull request by arnaubennassar**
### What does this PR does?

- Rebase with `develop`
- Split atomic related stuff from `api/txpool(_test).go` ==> `api/atomicpool(_test).go`
- Moved `common/formats.go` ==> `common/converters.go` (and deleted formats.go)

## Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Include tests
- [ ] Respect code style and lint
- [ ] Update swagger file (if needed)
- [ ] Update documentation (*.md) (if needed)

### CODE REVIEW

File: api/atomicpool.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+	const rqOffsetTwo = 2
+	const rqOffsetThree = 3
+	const rqOffsetFour = 4
+	const rqOffsetFive = 5
+	const rqOffsetSix = 6
+	const rqOffsetSeven = 7
+	const rqOffsetMinusFour = -4
+	const rqOffsetMinusThree = -3
+	const rqOffsetMinusTwo = -2
+	const rqOffsetMinusOne = -1
```
Comment (Mikelle) [inline api/atomicpool.go:121]:
```suggestion
	const (
	    rqOffsetMinusOne = -1
	    ...
	    rqOffsetMinusOne = -1
)	    
```

Comment (Mikelle) [reply api/atomicpool.go:121]: Maybe is it better to use a map structure instead of a function with multiple ifs

File: api/parsers/txpool.go [hunk 2]
```diff
@@ -28,6 +28,20 @@ func ParsePoolTxFilter(c *gin.Context) (common.TxID, error) {
...
+	if atomicGroupIDStr == "" {
+		return common.AtomicGroupID{}, tracerr.Wrap(fmt.Errorf("%s is required", name))
+	}
+	atomicGroupID, err := common.NewAtomicGroupIDFromString(atomicGroupIDStr)
+	if err != nil {
+		return common.AtomicGroupID{}, tracerr.Wrap(fmt.Errorf("invalid %s", name))
+	}
+	return atomicGroupID, nil
+}
+
```
Comment (Mikelle) [inline api/parsers/txpool.go:44]:
```suggestion
type AtomicGroupFilter struct {
  ID string `uri:"id" binding:"required"
}
// ParseParamAtomicGroupID func for parsing AtomicGroupID
func ParseParamAtomicGroupID(c *gin.Context) (common.AtomicGroupID, error) {
        var atomicGroupFilter AtomicGroupFilter
	if err := c.ShouldBindUri(&AtomicGroupFilter); err != nil {
	  return common.AtomicGroupID{}, traceerr.Wrap(err)
	}
	atomicGroupID, err := common.NewAtomicGroupIDFromString(atomicGroupFilter.ID)
	if err != nil {
		return common.AtomicGroupID{}, tracerr.Wrap(fmt.Errorf("invalid %s", name))
	}
	return atomicGroupID, nil
}
```

File: api/txspool.go [hunk 3]
```diff
@@ -85,192 +80,99 @@ func (a *API) getPoolTxs(c *gin.Context) {
...
+	}
+	if tx.RqFee != 0 {
+		panic("here")
+	}
+	if tx.RqNonce != 0 {
+		panic("here")
+	}
+	if tx.RqTokenID != 0 {
+		panic("here")
+	}
```
Comment (Mikelle) [inline api/txspool.go:176]: I think it's left after debugging

File: coordinator/coordinator_test.go [hunk 4]
```diff
@@ -115,12 +115,12 @@ func newTestModules(t *testing.T) modules {
 	var bjj babyjub.PublicKeyComp
 	err = bjj.UnmarshalText([]byte("c433f7a696b7aa3a5224efb3993baf0ccd9e92eecee0c29a3f6c8208a9e81d9e"))
 	require.NoError(t, err)
-	coordAccount := &txselector.CoordAccount{ // TODO TMP
+	coordAccount := txselector.CoordAccount{ // TODO TMP
```
Comment (Mikelle) [inline coordinator/coordinator_test.go:118]: I think, it will be better to give more descriptive todo message

Comment (arnaubennassar) [reply coordinator/coordinator_test.go:118]: Agree, unfortunately this message wasn't added by me and I don't know what it's supposed to mean

**Review by Mikelle — APPROVED** [review]

**Pull request merged**